README.md

```diff
@@ -144,33 +144,105 @@ $ make -C build.assets build-binaries
 
 ### Local Build
 
-To perform a build on your host, ensure you have installed Go. In order to
-include the Rust-powered features like Desktop Access, you'll
-also need `cargo` and `rustc`. The current versions of these tools can be found
-in `build.assets/Makefile`.
+**Important:**
 
-```bash
-# get the source & build:
+* The Go compiler is somewhat sensitive to the amount of memory: you will need
+  **at least** 1GB of virtual memory to compile Teleport. A 512MB instance
+  without swap will **not** work.
+* This will build the latest version of Teleport, **regardless** of whether it
+  is stable. If you want to build the latest stable release, run `git checkout`
+  to the corresponding tag (for example, run `git checkout v8.0.0`) **before**
+  performing a build.
+* The following instructions are Mac-specific.
+
+#### Dependencies
+
+Ensure you have installed correct versions of dependencies:
+
+* `Go` version from [go.mod](https://github.com/gravitational/teleport/blob/0e9d621b464f2bd05a28e7db0ae2ef65349c460e/go.mod#L3)
+  
+  ```shell
+  # if we are not on the latest, you might need to
+  # brew install go@<some_version>, i.e. 1.16
+  #
+  # check which version will be installed by running
+  # brew info go
+  
+  $ brew install go
+  ````
+  
+If you wish to build the Rust-powered features like Desktop Access:
+
+* `Rust` and `Cargo` version from [build.assets/Makefile](https://github.com/gravitational/teleport/blob/c3dde989cc9de6fa199f1a12287926218cc2e685/build.assets/Makefile#L21):
+  
+  ```shell
+  $ brew install rustup
+  $ rustup-init
+  # accept defaults
+  #
+  # Once command finishes successfully, you might need to add
+  # 
+  # export PATH="$HOME/.cargo/bin:$PATH"
+  # 
+  # into ~/.zprofile and run
+  # 
+  # $ . ~/.zprofile
+  # 
+  # or open a new shell
+  #
+  $ rustup toolchain install <version from build.assets/Makefile>
+  $ cd <teleport.git>
+  $ rustup override set <version from build.assets/Makefile>
+  $ rustc --version                                                                                                                                                                  ─╯
+  rustc <version from build.assets/Makefile> (db9d1b20b 2022-01-20)
+  ```
+To build `tsh` version > `10.x` with `libfido` support:  
+
+* Install `libfido2` and `openssl v 1.1.1` 
+  ```shell
+  $ brew install libfido2
+  $ brew install openssl@1.1
```

Comment:
Installing `openssl@1.1` explicitly is not necessary, but `export PKG_CONFIG_PATH="$(brew --prefix openssl@1.1)/lib/pkgconfig"` is.

As in other sections, I'd rather link folks to https://github.com/gravitational/teleport/blob/master/lib/auth/webauthncli/doc.go#L15 and https://github.com/Yubico/libfido2#installation, which have more complete instructions. We should also mention that it's necessary to pass `FIDO2=dynamic` to make for this to take effect.

```suggestion
  $ brew install libfido2
  $ export PKG_CONFIG_PATH="$(brew --prefix openssl@1.1)/lib/pkgconfig
  $ make build/tsh FIDO2=dynamic  # builds tsh with libfido2 support
```

Edit: I see you added some of this below, but some of the advice stands.

Reply:
Ah, I see, it pulls `openssl@1.1` as a dependency. Good point. And I am with you, will link to https://github.com/Yubico/libfido2#installation and https://github.com/gravitational/teleport/blob/master/lib/auth/webauthncli/doc.go#L15

> We should also mention that it's necessary to pass FIDO2=dynamic to make for this to take effect.

As in, explicitly in text, not just on the provided command line?

Reply:
Just the make command line is good! I wrote this comment too soon.